Samples/excel-add-in-chart-analyze-data/src/taskpane/taskpane.ts

```diff
@@ -0,0 +1,588 @@
+
+      //Show trendline
+      series.trendlines.add("Exponential");
+      let trendline = series.trendlines.getItem(0);
+      trendline.displayRSquared = true;
+
+      //High light title
+      var font = chart.title.getSubstring(21, 5).font;
+      var font2 = chart.title.getSubstring(31, 11).font;
+      font.color = "#FF7F50";
+      font2.color = "#FF7F50";
+
+      await context.sync();
+      showStatus('Success for "Sales/Leaflets correlation"', false);
+    });
+  } catch (error) {
+    showStatus(error, true);
+  }
+}
+
+//Add a ChartTitle to the first chart in the worksheet Sample
+async function showSalesTrendline() {
+  try {
+    await Excel.run(async (context) => {
+      showStatus("Running", false);
+      const sheet = context.workbook.worksheets.getItem("Sample");
+      let dataRange = sheet.getRange("H2:H33");
+      let chart = sheet.charts.add("Line", dataRange, "auto");
+      chart.activate();
+      chart.name = "overallSales";
+      chart.title.text = "Total Sales Trend in September";
+      chart.left = 900;
+      chart.top = 20;
+      chart.width = 300;
+      chart.setPosition("Q16", "V30");
+      chart.legend.visible = false;
+
+      let axis = chart.axes.categoryAxis;
+      let yaxis = chart.axes.valueAxis;
+      yaxis.displayUnit = "Thousands";
+      let categoryNameRange = sheet.getRange("A2:A33");
+      axis.setCategoryNames(categoryNameRange);
+
+      let trendlines = chart.series.getItemAt(0).trendlines;
+      trendlines.add("MovingAverage");
+      let tre = trendlines.getItem(0);
+      tre.movingAveragePeriod = 7;
+      showStatus('Success for "Overall sales trend"', false);
+    });
+  } catch (error) {
+    showStatus(error, true);
+  }
+}
+
+async function clearDatalables() {
+  try {
+    await Excel.run(async (context) => {
+      const sheet = context.workbook.worksheets.getItem("Sample");
+      let chart = sheet.charts.getItem("overallSales");
+      let points = chart.series.getItemAt(0).points;
+      points.load();
+      await context.sync();
+      let count = points.count;
+      for (let i = 0; i < count; i++) {
+        let point = points.getItemAt(i);
+        point.hasDataLabel = false;
+      }
+      await context.sync();
+    });
+  } catch (error) {
+    showStatus(error, true);
+  }
+}
+
+//add a Axis Title to the Category Axis of the the first chart in the worksheet
+async function highlightSales() {
+  try {
+    await Excel.run(async (context) => {
+      showStatus("Running", false);
+      const sheet = context.workbook.worksheets.getItem("Sample");
+      let chart = sheet.charts.getItem("overallSales");
+      let points = chart.series.getItemAt(0).points;
+      points.load();
+      await context.sync();
+      let count = points.count;
+      let max = 0;
+      //let index = 0;
+      for (let i = 0; i < count; i++) {
+        let point = points.getItemAt(i);
+        //point.hasDataLabel = false;
+        point.load("value");
+        await context.sync();
```

Comment:
Having a sync in a loop is a bad pattern to show. We have a whole article about that: https://learn.microsoft.com/en-us/office/dev/add-ins/concepts/correlated-objects-pattern. Could we rewrite this?